### ISSUE TITLE
defaultOptions is not configurable

### ISSUE THREAD

**Issue by igtm** [nuxt-community/apollo-module#161]
### What problem does this feature solve?

I'd like to set `defaultOptions` [like this](https://github.com/Akryum/vue-apollo/blob/f60cf4d4a594568ebf4e288cd083579bbce09b3a/docs/guide/apollo/README.md#special-options). 
But it does't seem to be configurable.

Any plan of adding this feature?

Thanks in advance.

<div align="right"><sub><em>This feature request is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/apollo-module/issues/c142">#c142</a>)</em></sub></div>

**Comment by dohomi** [comment]
I won't have time to implement it. But feel free to open a PR

**Comment by jesseschutt** [comment]
Is there currently a way to set `defaultOptions` via client configuration file?

**Comment by lin-yen** [comment]
This's works for me

```
  {
    httpEndpoint: 'http://xxxxxxxxxxxx',

    apollo: {
      defaultOptions: {
        query: {
          fetchPolicy: 'network-only'
        }
      }
    }
  }
```
> "@nuxtjs/apollo": "^4.0.0-rc2.3"

**Comment by ndan** [comment]
@lin-yen As I can see `vue-apollo` uses `provider.defaultOptions` instead of `client.defaultOptions` when building a smart query ([proof](https://github.com/Akryum/vue-apollo/blob/master/src/dollar-apollo.js#L91)). 

There is no way to change `provider.defaultOptions` via `nuxt.config.js` (at least now), so I added a plugin:
```
// plugins/apollo-overrides.js
export default ({ app }, inject) => {
  app.apolloProvider.defaultOptions = {
    $query: {
      fetchPolicy: 'no-cache'
    }
  }
}
```

Is there a better way to change `provider.defaultOptions`?

ps. I'm totally new to nuxt/vue and vue-apollo

**Comment by dohomi** [comment]
@ndan thats the preferred way for now 👍

**Comment by robsontenorio** [comment]
I think it should be as a plug-in because you can have many things that will bloat nuxt.config.js

**Comment by robsontenorio** [comment]
@dohomi Would you accept a PR for this? Would be very useful to set `defaultOptions` on `nuxtjs.config.js` 

https://www.apollographql.com/docs/react/api/apollo-client.html#apollo-client
```js
 watchQuery: {
    fetchPolicy: 'cache-and-network',
    errorPolicy: 'ignore',
  },
  query: {
    fetchPolicy: 'network-only',
    errorPolicy: 'all',
  },
  mutate: {
    errorPolicy: 'all'
  }
```

**Comment by dohomi** [comment]
@robsontenorio yes go ahead. Currently I'm very busy in job so I might some longer time to review. I also still wait for the Vue 2.6 release which will fix SSR issues of this package and then I hope the integration is smooth for all users

**Comment by kieusonlam** [comment]
Merged https://github.com/nuxt-community/apollo-module/pull/214

**Comment by gatherben** [comment]
First off, thanks to @robsontenorio for getting this in, I just wasted a good chunk of my day trying to disable apollo's cache and this thread helped quite a bit. However, it seems that while this PR fixed the core issue, the example provided in the readme didn't translate well into the the multi-client setup i have in my app. I tried adding the the defaultOptions to the main apollo config in nuxt.config.js, but found that they didn't show up when i inspected the particular client i was using. The solution was to add the default options to the client config like so...

nuxt.config.js
`
apollo: {
    clientConfigs: {
        default: '~/client-config.ts'
    },
},
`

client-config.ts
`
import { InMemoryCache } from "apollo-cache-inmemory";

export default function(){
	return {
		cache: new InMemoryCache(),
		apollo: {
			defaultOptions: {
				watchQuery: {
					fetchPolicy: 'no-cache',
				},
				query: {
					fetchPolicy: 'no-cache',
				},
				mutate: {
					fetchPolicy: 'no-cache',
				},
			},
		}
	}
}
`

Hopefully this helps the next person.
